state_change_block.py

```diff
@@ -33,21 +21,22 @@ class StateChange(StateChangeBase):
     a notification Signal.
     """
 
-    exclude = BoolProperty(default=True, title = "Exclude Existing Fields")
-    use_persistence = BoolProperty(title="Use Persistence", default=True,
-            visible=True)
-
-    def __init__(self):
-        super().__init__()
-        self._safe_lock = Lock()
-
-    def process_signals(self, signals):
-        self._logger.debug(
-            "Ready to process {} incoming signals".format(len(signals))
-        )
-        with self._safe_lock:
-            for signal in signals:
-                self._logger.debug("Attempting to set state")
-                out = self._process_state(signal, self.exclude)
-                if out is not None:
-                    self.notify_signals([out])
+    exclude = BoolProperty(default=True, title="Exclude Existing Fields")
+
+    def _process_group(self, signals, group, to_notify=list()):
+        """ Process the signals for a group.
+
+        Add any signals that should be passed through to the to_notify list
+        """
+        for signal in signals:
+            state_change = self._process_state(signal, group)
```

Comment:
The class docstring that says "*state* changing from None to not None does not count as a state change." is no longer valid.